qa/tasks/cephfs/fuse_mount.py

```diff
@@ -114,7 +114,17 @@ def _get_mount_cmd(self, mntopts):
         if self.client_keyring_path and self.client_id:
             mount_cmd += ['-k', self.client_keyring_path]
         if self.cephfs_mntpt:
```

Comment:
This configures which path in CephFS to mount. The QA suite should not specify both `cephfs_mntpt` **and** `mount_subvol_num`. You should confirm it is either one but not both (as that wouldn't make sense).

You should also save `path_to_mount` to `self.cephfs_mntpt`. Also suggest setting `cephfs_mntpt` to `/` if none of these configs are set (but continue to omit `--client_mountpoint=` if `/`).

Reply:
* added validation for `self.cephfs.mntpt` and `mount_subvol_num` and added assertion to force bail out if both are found
* also defaulting `self.cephfs_mntpt` to `'/'` if not set
* now setting `self.cephfs_mntpt` to `path_to_mount` if mounting a subvol

Reply:
@batrick a teuthology run with 10 jobs failed because `cephfs_mntpt` was set when the `subvolume` tests were executed
So, instead of aborting the `subvolume` tests when both `cephfs_mntpt` and `mount_subvol_num` are specified, I've updated the code to mount the subvolume while disregarding the `cephfs_mntpt`.

Does this sound reasonable ?

I also tried a `--filter-all` without an explicit `subvolume` argument string, but all of the 10 jobs chosen to run had one of the subvolume yamls chosen.
Does this mean that a subvolume yaml will always be chosen for an `fs:workload` job ?
Do I need to add/tweak somewhere to stop this from happening ?

Reply:
> Does this sound reasonable ?

If it's not possible to fix it, sure.

Reply:
@batrick FYI - Also, `qa/workunits/fs/misc/subvolume.sh` steps on the `fs:workload` subvolume configs and fails when exercising subvolume creation since the `fs:workload` subvolume config causes the subvolume to be mounted at `mnt.?`

Should the `qa/workunits/fs/misc/subvolume.sh` script always return success without running subvolume creation tests when the current directory is already a subvolume ?

The script `qa/workunits/fs/misc/subvolume.sh` was supposed be run on a barebones volume when exercising subvolume creation tests. Now that I've added an empty `no-subvolume.yaml` to the list of subvolume yamls, the shell script will be able to exercise subvolume creation tests with that particular yaml combination.

Reply:
> Should the qa/workunits/fs/misc/subvolume.sh script always return success without running subvolume creation tests when the current directory is already a subvolume ?

A cleaner solution may be to change `qa/suites/fs/workload/tasks/workunit/fs/misc.yaml` to override the mount so it's always at `/` rather than a subvolume. Otherwise, returning success from the script would be acceptable.